relay-profiling/src/sample.rs

```diff
@@ -226,6 +226,52 @@ impl SampleProfile {
         self.profile
             .strip_pointer_authentication_code(&self.platform, &self.device.architecture);
     }
+
+    fn remove_idle_samples_at_the_edge(&mut self) {
+        self.remove_idle_samples_at_the_start();
+        self.profile.samples.reverse();
+        self.remove_idle_samples_at_the_start();
+        self.profile.samples.reverse();
+    }
+
+    fn remove_idle_samples_at_the_start(&mut self) {
+        let mut thread_ids: HashSet<u64> = HashSet::new();
+        self.profile
+            .samples
+            .retain(|sample| match self.profile.stacks.get(sample.stack_id) {
+                Some(stack) => {
+                    if stack.is_empty() && !thread_ids.contains(&sample.thread_id) {
+                        false
+                    } else {
+                        thread_ids.insert(sample.thread_id);
+                        true
+                    }
+                }
+                None => true,
+            });
+    }
+
+    fn cleanup_thread_metadata(&mut self) {
+        if let Some(thread_metadata) = &mut self.profile.thread_metadata {
+            let mut thread_ids: HashSet<String> = HashSet::new();
+            for sample in &self.profile.samples {
+                thread_ids.insert(sample.thread_id.to_string());
```

Comment:
It's unfortunate that we have to convert to string here... could we change `thread_metadata` to contain `u64` thread IDs instead? Or would that fail parsing certain payloads?